### PR TITLE
test: Add direct-routing + L7 test

### PR DESCRIPTION
**Pull request by joestringer**

### CODE REVIEW

**Comment by joestringer** [comment]
test-focus with L7 policy

**Comment by joestringer** [comment]
test-focus K8s.*with L7 policy.*

**Comment by joestringer** [comment]
test-me-please

File: test/k8sT/DatapathConfiguration.go [hunk 1]
```diff
@@ -325,6 +325,30 @@ var _ = Describe("K8sDatapathConfig", func() {
 			}, DeployCiliumOptionsAndDNS)
 
 			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+
+			It("with L7 policy", func() {
```
Comment (brb) [inline test/k8sT/DatapathConfiguration.go]: `s/It/Context/`, as AFAIK `It` doesn't allow its body to contain setup routines. From the CI run of this PR:

```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.11-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:471
You may only call It from within a Describe, Context or When
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/k8s-1.11-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:471
```

**Review by brb — CHANGES_REQUESTED** [review]

**Comment by joestringer** [comment]
test-focus K8s.*with L7 policy.*

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Hit #11457 in K8s-1.17-Kernel-4.19 run.

**Comment by joestringer** [comment]
Hit #11442 in K8s-1.11-Kernel-netnext run.

**Review by brb — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Comment by joestringer** [comment]
I finally fixed my local testing environment and was unable to verify that L7 traffic was actually observed in this test as of the code today. #11625 will mainline the testing changes necessary to validate locally, and I'll need to respin this test to validate it actually works and tests L7 visibility.

**Pull request closed without merging**